rosbag2_compression/src/rosbag2_compression/sequential_compression_reader.cpp

```diff
@@ -43,64 +44,66 @@ SequentialCompressionReader::~SequentialCompressionReader()
 
 void SequentialCompressionReader::setup_decompression()
 {
-  compression_mode_ = rosbag2_compression::compression_mode_from_string(metadata_.compression_mode);
+  if (decompressor_) {
+    return;
+  }
+
+  compression_mode_ = compression_mode_from_string(metadata_.compression_mode);
   if (compression_mode_ != rosbag2_compression::CompressionMode::NONE) {
     decompressor_ = compression_factory_->create_decompressor(metadata_.compression_format);
-    // Decompress the first file so that it is readable; don't need to do anything for
-    // per-message encryption.
-    if (compression_mode_ == CompressionMode::FILE) {
-      ROSBAG2_COMPRESSION_LOG_DEBUG_STREAM("Decompressing " << get_current_file().c_str());
-      *current_file_iterator_ = decompressor_->decompress_uri(get_current_file());
-    }
   } else {
     throw std::invalid_argument{
             "SequentialCompressionReader requires a CompressionMode that is not NONE!"};
   }
+  if (!decompressor_) {
```

Comment:
Would you want to simplify the "if (condition) then throw" with this?
```c++
rcpputils::require_true(decompressor_ != nullptr, "Could't initialize decompressor.")
```
This will change the thrown error, but the documentation doesn't call out that `std::runtime_error` is thrown anyway.